tests/rhtap-demo/rhtap-demo.go

```diff
@@ -94,648 +94,649 @@ var _ = framework.RhtapDemoSuiteDescribe(Label("rhtap-demo"), Label("verify-stag
-					GinkgoWriter.Printf("Parallel processes: %d\n", suiteConfig.ParallelTotal)
-					GinkgoWriter.Printf("Running on namespace: %s\n", namespace)
-					GinkgoWriter.Printf("User: %s\n", fw.UserName)
-				})
-
-				// Remove all resources created by the tests
-				AfterAll(func() {
-					if !appTest.Stage {
-						// collect SPI ResourceQuota metrics (temporary)
-						err := fw.AsKubeAdmin.CommonController.GetResourceQuotaInfo("rhtap-demo", namespace, "appstudio-crds-spi")
-						Expect(err).NotTo(HaveOccurred())
-
-						if !(strings.EqualFold(utils.GetEnv("E2E_SKIP_CLEANUP", ""), "true") && !CurrentSpecReport().Failed()) {
-							// RHTAPBUGS-978: temporary timeout to 15min
-							if err := fw.AsKubeAdmin.HasController.DeleteAllComponentsInASpecificNamespace(namespace, 15*time.Minute); err != nil {
-								if err := fw.AsKubeAdmin.StoreAllArtifactsForNamespace(namespace); err != nil {
-									Fail(fmt.Sprintf("error archiving artifacts:\n%s", err))
-								}
-								Fail(fmt.Sprintf("error deleting all componentns in namespace:\n%s", err))
-							}
-							Expect(fw.AsKubeAdmin.HasController.DeleteAllApplicationsInASpecificNamespace(namespace, 30*time.Second)).To(Succeed())
-							Expect(fw.AsKubeAdmin.CommonController.DeleteAllSnapshotEnvBindingsInASpecificNamespace(namespace, 30*time.Second)).To(Succeed())
-							Expect(fw.AsKubeAdmin.IntegrationController.DeleteAllSnapshotsInASpecificNamespace(namespace, 30*time.Second)).To(Succeed())
-							Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllEnvironmentsInASpecificNamespace(namespace, 30*time.Second)).To(Succeed())
-							Expect(fw.AsKubeAdmin.TektonController.DeleteAllPipelineRunsInASpecificNamespace(namespace)).To(Succeed())
-							Expect(fw.AsKubeAdmin.GitOpsController.DeleteAllGitOpsDeploymentsInASpecificNamespace(namespace, 30*time.Second)).To(Succeed())
-							Expect(fw.SandboxController.DeleteUserSignup(fw.UserName)).To(BeTrue())
-						}
-					} else {
-						err := fw.AsKubeDeveloper.HasController.DeleteAllApplicationsInASpecificNamespace(fw.UserNamespace, stageTimeout)
-						if err != nil {
-							GinkgoWriter.Println("Error while deleting resources for user, got error: %v\n", err)
-						}
-						Expect(err).NotTo(HaveOccurred())
-						err = fw.AsKubeDeveloper.HasController.DeleteAllComponentDetectionQueriesInASpecificNamespace(fw.UserNamespace, stageTimeout)
-						if err != nil {
-							GinkgoWriter.Println("while deleting component detection queries for user, got error: %v\n", err)
+	Describe("rhtap-demo E2E test", Ordered, func() {
+		BeforeAll(func() {
+			if Label("verify-stage").MatchesLabelFilter(GinkgoLabelFilter()) {
+				token = utils.GetEnv("STAGEUSER_TOKEN", "")
+				ssourl = utils.GetEnv("STAGE_SSOURL", "")
+				apiurl = utils.GetEnv("STAGE_APIURL", "")
+				username = utils.GetEnv("STAGE_USERNAME", "")
+				if token == "" && ssourl == "" && apiurl == "" && username == "" {
+					Fail("Failed: Please set the required Stage Variables for user")
+				}
```

Comment:
I think only this part (which is problematic) should be moved to BeforeAll, otherwise the test cases for rhtap-demo and verify-stage won't be generated at all

Reply:
Done, PTAL